### PR TITLE
GH-2850 Add FATE support for BLS12-381 crypto operations

### PR DESCRIPTION
**Pull request by hanssv**
Depends on https://github.com/aeternity/aebytecode/pull/88
and https://github.com/aeternity/aesophia/pull/187

Refs #2850

### CODE REVIEW

**Comment by hanssv** [comment]
The `emcl` dependency doesn't yet build on Windows, so WIP. cc @tolbrino

File: .circleci/config.yml [hunk 1]
```diff
@@ -863,6 +863,8 @@ jobs:
             brew install libsodium erlang@20
             brew link libsodium
             brew link --force erlang@20
+            brew install gmp erlang@20
```
Comment (dincho) [inline .circleci/config.yml]: why not adding it to the install row above ? this looks like copy-paste nonsense to me 🤔

File: .circleci/config.yml [hunk 2]
```diff
@@ -863,6 +863,8 @@ jobs:
...
+            brew link --force gmp erlang@20
```
Comment (dincho) [inline .circleci/config.yml]: erlang is already linked above, also do you really need force?

File: Dockerfile [hunk 3]
```diff
@@ -14,7 +14,7 @@ FROM ubuntu:18.04
 COPY --from=builder /app/_build/prod/rel/aeternity /home/aeternity/node
 
 # OpenSSL is shared lib dependency
-RUN apt-get -qq update && apt-get -qq -y install libssl1.0.0 curl libsodium23 \
+RUN apt-get -qq update && apt-get -qq -y install libssl1.0.0 curl libsodium23 libgmp-dev \
```
Comment (dincho) [inline Dockerfile]: could please try a runtime package only without -dev?

Comment (hanssv) [reply Dockerfile]: @dincho I blame @tolbrino for everything, just doing what he told me :-)

Comment (hanssv) [reply Dockerfile]: I guess it should be `libgmp10` then? 🤔

Comment (hanssv) [reply Dockerfile]: Fixed

**Pull request merged**